packages/admin/admin/src/table/filterbar/filterBarButton/FilterBarButton.tsx

```diff
@@ -1,59 +1,119 @@
+import { ThemedComponentBaseProps } from "helpers/ThemedComponentBaseProps";
 import * as React from "react";
 
 import { FilterBarActiveFilterBadge, FilterBarActiveFilterBadgeProps } from "../filterBarActiveFilterBadge/FilterBarActiveFilterBadge";
-import { FilterBarButtonClassKey, styles } from "./FilterBarButton.styles";
 
 /**
  * @deprecated Use MUI X Data Grid in combination with `useDataGridRemote` instead.
  */
-export interface FilterBarButtonProps extends ButtonProps {
+
+export type FilterBarButtonClassKey = "root" | "open" | "hasDirtyFields" | "filterBadge";
+
+type OwnerState = { hasDirtyFields: boolean; openPopover: boolean | undefined };
+
+const Root = styled(Button, {
+    name: "CometAdminFilterBarButton",
+    slot: "root",
+    overridesResolver({ ownerState }: { ownerState: OwnerState }, styles) {
+        return [styles.root, ownerState.openPopover && styles.open, ownerState.hasDirtyFields && styles.hasDirtyFields];
+    },
+})<{ ownerState: OwnerState }>(
+    ({ theme, ownerState }) => css`
+        position: relative;
+        cursor: pointer;
+        display: flex;
+        background-color: ${theme.palette.common.white};
+        border-color: ${theme.palette.grey[100]};
+        border-radius: 2px;
+
+        && .${buttonClasses.startIcon} .${svgIconClasses.root}, && .${buttonClasses.endIcon} .${svgIconClasses.root} {
+            font-size: 12px;
+        }
+
+        &:hover {
+            border-color: ${theme.palette.grey[100]};
+            background-color: ${theme.palette.common.white};
+        }
+
+        &:focus {
+            border-color: ${theme.palette.primary.main};
+            background-color: ${theme.palette.common.white};
+        }
+
+        ${ownerState.openPopover &&
+        css`
+            border-color: ${theme.palette.primary.main};
+        `}
+
+        ${ownerState.hasDirtyFields &&
+        css`
+            border-color: ${theme.palette.grey[400]};
+            font-weight: ${theme.typography.fontWeightBold};
+
+            &:disabled {
+                border-color: ${theme.palette.grey[100]};
+            }
+        `}
+    `,
+);
+
+const FilterBadge = styled("span", {
+    name: "CometAdminFilterBarButton",
+    slot: "filterBadge",
+    overridesResolver(_, styles) {
+        return [styles.filterBadge];
+    },
+})(
+    css`
+        margin-left: 6px;
+    `,
+);
+
+export interface FilterBarButtonProps extends ThemedComponentBaseProps<{ root: typeof Button; filterBadge: "span" }>, ButtonProps {
     dirtyFieldsBadge?: React.ComponentType<FilterBarActiveFilterBadgeProps>;
     numberDirtyFields?: number;
     openPopover?: boolean;
 }
 
-const FilterBarButton = ({
-    children,
-    dirtyFieldsBadge,
-    numberDirtyFields,
-    openPopover,
-    classes,
-    endIcon = <ChevronDown />,
-    className,
-    ...buttonProps
-}: FilterBarButtonProps & WithStyles<typeof styles>): React.ReactElement => {
+export function FilterBarButtonWithStyles(inProps: FilterBarButtonProps) {
```

Comment:
This should just be named `FilterBarButton` then the `export { FilterBarButtonWithStyles as FilterBarButton }` below can be removed.